platforms/android/tangram/src/main/java/com/mapzen/tangram/MapView.java

```diff
@@ -170,18 +171,12 @@ public void getMapAsync(@NonNull final MapReadyCallback readyCallback,
                             @Nullable final HttpHandler handler) {
 
         disposeMapReadyTask();
-        if (glSurfaceView == null) {
-            glSurfaceView = new GLSurfaceView(getContext());
-        }
-        executeMapAsyncTask(readyCallback, handler);
-    }
 
-    @NonNull
-    protected MapController getMapInstance(@Nullable final HttpHandler handler) {
-        return new MapController(glSurfaceView, handler);
+        executeMapAsyncTask(readyCallback, handler);
     }
 
     protected void configureGLSurfaceView() {
+        glSurfaceView = new GLSurfaceView(getContext());
```

Comment:
Does it make sense to only init a glSurfaceView once, like I had it before?

Reply:
Maybe I am wrong, but the reason I had `GLSurfaceView` created before initializing `MapController` was to make sure `Context` is passed post initialization for `GLSurfaceView` to account for any MapView context changes happening because of `GLSurfaceView`.